### PR TITLE
`react-redux` `ComponentDecorator` input accepts `TOwnProps`

### PR DESCRIPTION
**Pull request by thasner**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/thasner/DefinitelyTyped/blob/d5f4b6b31de47f4787ec8d58428793edccbd4f9b/types/react-redux/react-redux-tests.tsx#L362
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

File: types/react-redux/index.d.ts
Comment (thasner) [inline types/react-redux/index.d.ts]: In the old version, `TOriginalProps` was `TStateProps & TDispatchProps`. This meant that the `component` parameter had type `Component<TStateProps & TDispatchProps>`.

In the new version, `component` is properly `Component<TStateProps & TDispatchProps & TOwnProps>`

File: types/react-redux/react-redux-tests.tsx
Comment (thasner) [inline types/react-redux/react-redux-tests.tsx]: the type `{}` allowed the compiler to overlook the missing `TOwnProps`, since `{}` is basically `any`.

Comment (seansfkelley) [inline types/react-redux/react-redux-tests.tsx]: Indeed: #8787

File: types/react-redux/index.d.ts
Comment (seansfkelley) [inline types/react-redux/index.d.ts]: Does this still work as intended, i.e., does it properly infer `IOwnProps`? Also, please rename it to `TOwnProps` for consistency.

**Review by seansfkelley — CHANGES_REQUESTED** [review]

File: types/react-redux/index.d.ts
Comment (thasner) [inline types/react-redux/index.d.ts]: I originally tried to remove this, and failed the tests https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/222996842

this format is equivalent to the previous. the usage of `T extends Component<IOwnProps>` was the important bit for some of the tests.

yes, `IOwnProps` should be `TOwnProps`, sorry :/
fixing that now

Comment (thasner) [inline types/react-redux/index.d.ts]: this test is relevant https://github.com/DefinitelyTyped/DefinitelyTyped/blob/dae4a527d7a48735bfb3b4fd6c1e59a400b4cb6d/types/react-redux/react-redux-tests.tsx#L282

**Review by seansfkelley — APPROVED** [review]

**Comment by seansfkelley** [comment]
@RyanCavanaugh revision needed? I think this is good to go. More generally, do you maintainers see the github reviews or the thumbs emojis as the final LGTM indicator?

**Comment by RyanCavanaugh** [comment]
There's a bug in my script here, it shouldn't have the RN tag after your approved review.

The script to set the tags sees :+1: on the dt-bot comment, a comment containing :+1:, or a comment containing "LGTM" as "Approval"

**Comment by seansfkelley** [comment]
Cool, thanks for the clarification!

File: types/react-redux/index.d.ts
Comment (d-i-bondarenko) [inline types/react-redux/index.d.ts]: @thasner But it's not always that way. I mean `Component<TStateProps & TDispatchProps & TOwnProps>` can be without TOwnProps at all if you use `connect` with `mergeProps` parameter. `mergeProps` returns `TStateProps & TDispatchProps`, so it seems that only this two props are required in returned Component by `connect`.

Comment (thasner) [inline types/react-redux/index.d.ts]: Hey @d-i-bondarenko , I just saw https://github.com/DefinitelyTyped/DefinitelyTyped/issues/16021, so I'll continue the conversation there

**Pull request merged**
